### PR TITLE
Fix AudioListener2D ignoring rotation for positional audio

### LINKED ISSUE #69025: `AudioListener2D` ignores rotation for positional audio
### Godot version

v4.0.beta5.official [89a33d28f]

### System information

Windows 10

### Issue description

When using an `AudioListener2D` with `AudioStreamPlayer2D`, only the position is taken into account for the positional audio calculation, but rotation is ignored.
Without and active `AudioListener2D`, the positional audio works fine both when changing position and rotation.

### Steps to reproduce

Open and run the attached reproduction project.
Move around by dragging with left mouse button, rotate by dragging left/right with the right mouse button.
Pressing Space enables/disables `AudioListener2D`.

Notice that when the listener is disabled, both moving and rotating sounds fine, but with it enabled, rotation is ignored (which leads to some very weird audio when the listener has both been moved and rotated)

### Minimal reproduction project

[AudioListener2DTest.zip](https://github.com/godotengine/godot/files/10070958/AudioListener2DTest.zip)

### PR DESCRIPTION
**Pull request by Uxeron**
Fixes #69025.

### CODE REVIEW

File: scene/2d/audio_stream_player_2d.cpp [hunk 1]
```diff
@@ -174,13 +174,14 @@ void AudioStreamPlayer2D::_update_panning() {
 
 		//screen in global is used for attenuation
 		AudioListener2D *listener = vp->get_audio_listener_2d();
+		Transform2D full_canvas_transform = vp->get_global_canvas_transform() * vp->get_canvas_transform();
 		if (listener) {
 			listener_in_global = listener->get_global_position();
-			relative_to_listener = global_pos - listener_in_global;
+			relative_to_listener = (global_pos - listener_in_global).rotated(-listener->get_global_rotation());
+			relative_to_listener *= full_canvas_transform.get_scale(); // Default listener scales with canvas size, do the same here
```
Comment (KoBeWi) [inline scene/2d/audio_stream_player_2d.cpp:181]:
```suggestion
			relative_to_listener *= full_canvas_transform.get_scale(); // Default listener scales with canvas size, do the same here.
```

File: scene/2d/audio_stream_player_2d.cpp [hunk 2]
```diff
@@ -174,13 +174,14 @@ void AudioStreamPlayer2D::_update_panning() {
 
 		//screen in global is used for attenuation
 		AudioListener2D *listener = vp->get_audio_listener_2d();
+		Transform2D full_canvas_transform = vp->get_global_canvas_transform() * vp->get_canvas_transform();
```
Comment (KoBeWi) [inline scene/2d/audio_stream_player_2d.cpp:177]: Not sure if this transform is correct in all cases.

**Review by KoBeWi — APPROVED** [review]

**Review by ellenhp — APPROVED** [review]
No significant audio changes so I feel comfortable approving as long as it's approved for 2d-nodes and tests correctly. The complexity here is transform math which I always struggle to reason about. Thank you for the contribution. I believe this may be my oversight originally?

File: scene/2d/audio_stream_player_2d.cpp (hunk 1 above)
Comment (Uxeron) [reply scene/2d/audio_stream_player_2d.cpp]: Fixed comment.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**